### ISSUE TITLE
[Feature Request] Better diff between expected and actual in blocTest

### ISSUE THREAD

**Issue by Albert221** [felangel/bloc#1783]
**Is your feature request related to a problem? Please describe.**
Right now when my state differs from what I expect in bloc tests I get a big load of text on my screen showing what expected and actual states look like. The problem is, sometimes my state contains many items or the item itself is very big. It's **REALLY** hard to read it.

**Describe the solution you'd like**
I can see two solutions:
- Showing a diff as GitHub does.
- Outputting only the parts of the object that really differed. So that when I have some object as a state, but only 2 out of 25 fields differ, I'd only care to see those two fields' values instead of the whole object.

**Describe alternatives you've considered**
Right now I just copy&paste both expected and actual states to some diff checker online to see the difference.

**Additional context**
Find difference with your eye :)

![image](https://user-images.githubusercontent.com/4085280/94671393-1d419a80-0314-11eb-825a-21b8efa5d5b8.png)

**Comment by felangel** [comment]
I made some progress on this and wanted to gather feedback. I decided to stay consistent with how `git diff --word-diff=plain` represents differences.

In the below example, the expected output is a post with an id 1 and the actual was a post with an id of 0.

![Screen Shot 2021-11-04 at 12 00 12 AM](https://user-images.githubusercontent.com/8855632/140260495-4fe463a1-cd46-4261-b73f-3cc24130b0b5.png)

If you have any feedback let me know 👍
